src/main/java/software/amazon/smithy/lsp/ext/LspLog.java

```diff
@@ -90,18 +92,33 @@ private static String currentTime() {
      * @param message object to write, will be converted to String
      */
     public static void println(Object message) {
-        String sanitizedMessage = message != null ? message.toString() : "null";
+        String sanitizedMessage = processMessage(message);
         String timestamped = "[" + currentTime() + "] " + sanitizedMessage;
         try {
             if (fw != null) {
                 fw.append(timestamped + "\n").flush();
             } else {
                 synchronized (buffer) {
-                    buffer.ifPresent(buf -> buf.add(timestamped));
+                    buffer.ifPresent(buf -> buf.add(sanitizedMessage));
                 }
             }
         } catch (Exception e) {
 
         }
     }
+
+    private static String processMessage(Object message) {
+        if (message == null) {
+            return "null";
+        } else {
+            if (message instanceof Throwable) {
+                StringWriter sw = new StringWriter();
+                PrintWriter pw = new PrintWriter(sw);
+                ((Throwable) message).printStackTrace(pw);
+                return sw.toString();
+            } else {
+                return message.toString();
+            }
+        }
```

Comment:
```suggestion
        if (message == null) {
            return "null";
        } else if (message instanceof Throwable) {
            StringWriter sw = new StringWriter();
            PrintWriter pw = new PrintWriter(sw);
            ((Throwable) message).printStackTrace(pw);
            return sw.toString();
        } else {
            return message.toString();
        }
```
I could be missing something, but I believe this would work the same.

Reply:
absolutely